Add _reshard HTTP API reference documentation (#404)
apache/couchdb-documentation: src/config/resharding.rst

..
..   http://www.apache.org/licenses/LICENSE-2.0
..
.. Unless required by applicable law or agreed to in writing, software
.. distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
.. WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
.. License for the specific language governing permissions and limitations under
.. the License.

.. highlight:: ini

==========
Resharding
==========

.. _config/reshard:

Resharding Configuration
========================

.. config:section:: resharding :: Resharding Configuration

    .. config:option:: max_jobs

        Maximum number of resharding jobs per cluster node. This includes
        completed, failed, and running jobs. If the job appears in the
        _reshard/jobs HTTP API results it will be counted towards the limit.
        When more than ``max_jobs`` jobs have been created, subsequent requests
        will start to fail with the ``max_jobs_exceeded`` error::

             [reshard]
             max_jobs = 25

    .. config:option:: max_retries

        How many times to retry shard splitting steps if they fail. For
        example, if indexing or topping off fails, it will be retried up to

>         these many times before the whole resharding job fails::

Reviewer: ```suggestion
        this many times before the whole resharding job fails::
```

             [reshard]
             max_retries = 1

    .. config:option:: retry_interval_sec

>         How to long to wait between subsequent retries::

Reviewer: ```suggestion
        How long to wait between subsequent retries::
```

             [reshard]
             retry_interval_sec = 10

    .. config:option:: delete_source

        Indicates if the source shard should be deleted after resharding has
        finished. By default, it is ``true`` as that would recover the space
        utilized by the shard. When debugging or when extra safety is required,
        this can be switched to ``false``::

             [reshard]
             delete_source = true

    .. config:option:: update_shard_map_timeout_sec

        How many seconds to wait for the shard map update operation to

>         complete. If there are a large number of shards db change waiting to

Reviewer: ```suggestion
        complete. If there is a large number of db shard changes waiting to
```
